packages/kbn-es/src/integration_tests/cluster.test.ts

```diff
@@ -0,0 +1,815 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import { ToolingLog, ToolingLogCollectingWriter } from '@kbn/tooling-log';
+import * as extractConfig from '../utils/extract_config_files';
+import * as dockerUtils from '../utils/docker';
+import { createAnyInstanceSerializer, createStripAnsiSerializer } from '@kbn/jest-serializers';
+import * as installUtils from '../install';
+import { Cluster } from '../cluster';
+import { ES_NOPASSWORD_P12_PATH } from '@kbn/dev-utils/src/certs';
+import {
+  DownloadSnapshotOptions,
+  InstallArchiveOptions,
+  InstallSnapshotOptions,
+  InstallSourceOptions,
+} from '../install/types';
+
+expect.addSnapshotSerializer(createAnyInstanceSerializer(ToolingLog));
+expect.addSnapshotSerializer(createStripAnsiSerializer());
+
+const log = new ToolingLog();
+const logWriter = new ToolingLogCollectingWriter();
```

Comment:
Interesting, what does this do?

Reply:
We use it to validate the logger output during specific calls, e.g. here example for `cluster.downloadSnapshot`:

```
    await cluster.downloadSnapshot(options);
    ...
    expect(logWriter.messages).toMatchInlineSnapshot(`
      Array [
        " info source[@kbn/es Cluster] Downloading snapshot",
      ]
    `);
```

It was added by Spencer, I think it is cool :)